app/models/spotlight/resource.rb

```diff
@@ -74,72 +35,141 @@ def needs_provider?
       type.blank?
     end
 
-    def save_and_commit
-      save.tap do
-        begin
-          blacklight_solr.commit
-        rescue => e
-          Rails.logger.warn "Unable to commit to solr: #{e}"
+    ##
+    # Persist the record to the database, and trigger a reindex to solr
+    #
+    # @param [Hash] All arguments will be passed through to ActiveRecord's #save method
+    def save_and_index(*args)
+      save(*args).tap { reindex }
+    end
+
+    concerning :GeneratingSolrDocuments do
+      ##
+      # @abstract
+      # Convert this resource into zero-to-many new solr documents. The data here
+      # should be merged into the resource-specific {#to_solr} data.
+      #
+      # @return [Hash] a single solr document hash
+      # @return [Enumerator<Hash>] multiple solr document hashes. This can be a
+      #   simple array, or an lazy enumerator
+      def to_solr
+        (exhibit_specific_solr_data || {}).merge(spotlight_resource_metadata_for_solr || {})
+      end
+
+      protected
+
+      ##
+      # @return an enumerator of all the indexable documents for this resource
+      def documents_to_index
+        data = to_solr
+        return [] if data.blank?
+        data &&= [data] if data.is_a? Hash
+
+        return to_enum(:documents_to_index) { data.size } unless block_given?
+
+        data.reject(&:blank?).each do |doc|
+          yield doc.reverse_merge(existing_solr_doc_hash(doc[unique_key]) || {})
         end
       end
-    end
 
-    protected
+      private
 
-    def blacklight_solr
-      @solr ||= RSolr.connect(connection_config)
-    end
+      ##
+      # Get any exhibit-specific metadata stored in e.g. sidecars, tags, etc
```

Comment:
Did this comment line get separated from the right method (def exhibit_specific_solr_data) ?